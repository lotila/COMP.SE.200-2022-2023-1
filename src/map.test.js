import map from "./map.js";

test("square list of numbers", () => {
  const square = (n) => n * n;

  expect(map([4, 8], square)).toStrictEqual([16, 64]);
});

test("Transform name-properties uppercase for list of objects", () => {
  const toUpper = ({ name }) => name.toUpperCase();

  expect(map([{ id: 1234, name: "item1" }, { id: 4321, name: "item2" },], toUpper)).toMatchSnapshot();
});

test("map over empty array", () => {
  expect(map([], null)).toStrictEqual([]);
});

test("use map to reverse order of items in array", () => {
  const reverse = (_, index, array) => array[array.length - 1 - index]
  expect(map([1, 2, 3, 4], reverse)).toStrictEqual([4, 3, 2, 1]);
});
