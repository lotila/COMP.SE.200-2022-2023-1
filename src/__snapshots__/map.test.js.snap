// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transform name-properties uppercase for list of objects 1`] = `
[
  "ITEM1",
  "ITEM2",
]
`;
